packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-group.tsx

```diff
@@ -0,0 +1,203 @@
+
+  @State()
+  private processedAvatars: ProcessedAvatar[] = [];
+
+  get countAvatars() {
+    return this.avatarList.length ?? 0;
+  }
+
+  async componentWillLoad(): Promise<void> {
+    this.processAvatars();
+
+    trackComponent(this.root);
+  }
+
+  disconnectedCallback() {
+    if (this.listItemClickListeners.size > 0) {
+      this.removeAllListeners();
+    }
+  }
+
+  private getAccent(index): string {
+    return `${(index % 12) + 1}` as GuxAvatarAccent;
+  }
+
+  private processAvatars(): void {
+    const avatarElements = Array.from(
+      this.root.children
+    ) as HTMLGuxAvatarFocusableBetaElement[];
+
+    this.processedAvatars = avatarElements.map((focusableAvatar, index) => {
+      const avatar = focusableAvatar.querySelector(
+        'gux-avatar-beta'
+      ) as HTMLGuxAvatarBetaElement;
+      const button = focusableAvatar.querySelector('button');
+      const link = focusableAvatar.querySelector('a');
+      const img = avatar.querySelector('img');
+      return {
+        focusableAvatar,
+        avatar,
+        button,
+        link,
+        img,
+        name: avatar.name,
+        accent: this.getAccent(index),
+        isOverflow: index >= this.avatarLimit
+      } as ProcessedAvatar;
+    });
+  }
+
+  private renderAvatar(
+    avatar: ProcessedAvatar,
+    avatarSize: GuxAvatarSize
+  ): JSX.Element {
+    if (avatar.button) {
+      return (
+        <button type={avatar.button.type} onClick={avatar.button.onclick}>
+          <gux-avatar-beta
+            size={avatarSize}
+            name={avatar.name}
+            accent={avatar.accent}
+          >
+            {avatar.img ? (
+              <img slot="image" src={avatar.img.src} alt={avatar.img.alt} />
+            ) : null}
+          </gux-avatar-beta>
+        </button>
+      ) as JSX.Element;
+    } else if (avatar.link) {
+      return (
+        <a href={avatar.link.href} target={avatar.link.target}>
+          <gux-avatar-beta
+            size={avatarSize}
+            name={avatar.name}
+            accent={avatar.accent}
+          >
+            {avatar.img ? (
+              <img slot="image" src={avatar.img.src} alt={avatar.img.alt} />
+            ) : null}
+          </gux-avatar-beta>
+        </a>
+      ) as JSX.Element;
+    } else {
+      return (
+        <gux-avatar-beta
+          size={avatarSize}
+          name={avatar.name}
+          accent={avatar.accent}
+        >
+          {avatar.img ? (
+            <img slot="image" src={avatar.img.src} alt={avatar.img.alt} />
+          ) : null}
+        </gux-avatar-beta>
+      ) as JSX.Element;
+    }
+  }
+
+  private renderListItem(avatar: ProcessedAvatar): JSX.Element {
+    const handleClick = (event: MouseEvent) =>
+      this.handleListItemClick(event, avatar);
+    this.listItemClickListeners.set(avatar.focusableAvatar, handleClick);
+
+    return (
+      <gux-list-item key={avatar.name} onClick={handleClick}>
+        <div class="gux-overflow-list-item">
```

Comment:
I think this will need to be a `span` element as the `gux-list-item` wraps the slotted content in a `button` tag and slotting a `div` wont be valid HTML.